namerd/iface/mesh/src/main/scala/io/buoyant/namerd/iface/MeshIfaceInitializer.scala

```diff
@@ -2,17 +2,67 @@ package io.buoyant.namerd
 package iface
 
 import com.fasterxml.jackson.annotation.JsonIgnore
-import com.twitter.finagle.{Namer, Path, Stack}
+import com.twitter.finagle._
 import com.twitter.finagle.buoyant.H2
+import com.twitter.finagle.buoyant.h2.AccessLogger
 import com.twitter.finagle.netty4.ssl.server.Netty4ServerEngineFactory
 import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.finagle.tracing.NullTracer
+import com.twitter.logging._
+import com.twitter.util.{Return, Throw, Try}
 import io.buoyant.grpc.runtime.ServerDispatcher
 import io.netty.handler.ssl.ApplicationProtocolNames
 import java.net.{InetAddress, InetSocketAddress}
 
-import com.twitter.finagle.tracing.NullTracer
-
 class MeshIfaceConfig extends InterfaceConfig {
+
+  var h2AccessLog: Option[String] = None
+  var h2AccessLogRollPolicy: Option[String] = None
+  var h2AccessLogAppend: Option[Boolean] = None
+  var h2AccessLogRotateCount: Option[Int] = None
+
+  @JsonIgnore
+  def mkAccessLogger(
+    logFilePath: Option[String],
+    logRollPolicy: Option[String],
+    logAppend: Option[Boolean],
+    logRotateCount: Option[Int]
+  ): Option[AccessLogger] = {
+    val policy = logRollPolicy match {
+      case None => Policy.Never
+      case Some(policyName) =>
+        Try(Policy.parse(policyName)) match {
+          // Default to Never roll policy if we can't parse one from the provided string
+          case Throw(_) => Policy.Never
+          case Return(r) => r
+        }
+    }
+    val append = logAppend.getOrElse(true)
+    val logRotate = logRotateCount.getOrElse(-1)
+
+    val filter = logFilePath match {
```

Comment:
why save this to a val and then immediately return it?

Reply:
I think it may be left over from me debugging this piece of code. Good catch.